lib/internal/http2/core.js

```diff
@@ -1479,7 +1480,15 @@ class ClientHttp2Session extends Http2Session {
 
     const onConnect = requestOnConnect.bind(stream, headersList, options);
     if (this.connecting) {
-      this.once('connect', onConnect);
+      if (this[kPendingRequestCalls]) {
```

Comment:
Can you initialize this in the constructor, e.g. as `null`, and then check with `!== null` here? Generally, JS engines, including V8, can sometimes perform better when all fields are present from the beginning.